src/mscorlib/src/System/Char.cs

```diff
@@ -234,7 +234,7 @@ public bool Equals(Char obj)
           if (IsLatin1(c)) {
               if (IsAscii(c)) {
```

Comment:
The `IsAscii` check should probably be moved before the `IsLatin1` check, since all ASCII chars are Latin1 chars and I'm assuming ASCII input is going to be more common in these methods.